PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationBasicPanel.java

```diff
@@ -0,0 +1,113 @@
+
+    private JPanel rootPanel;
+    @Getter
+    private SqlServerNameTextField serverNameTextField;
+    @Getter
+    private AdminUsernameTextField adminUsernameTextField;
+    @Getter
+    private JPasswordField passwordField;
+    @Getter
+    private JPasswordField confirmPasswordField;
+
+    private AzurePasswordFieldInput passwordFieldInput;
+    private AzurePasswordFieldInput confirmPasswordFieldInput;
+
+    private SqlServerConfig config;
+
+    SqlServerCreationBasicPanel(SqlServerConfig config) {
+        super();
+        this.config = config;
+        $$$setupUI$$$(); // tell IntelliJ to call createUIComponents() here.
+        init();
+        initListeners();
+        setData(config);
+    }
+
+    private void init() {
+        serverNameTextField.setSubscription(config.getSubscription());
+        passwordFieldInput = PasswordUtils.generatePasswordFieldInput(this.passwordField, this.adminUsernameTextField);
+        confirmPasswordFieldInput = PasswordUtils.generateConfirmPasswordFieldInput(this.confirmPasswordField, this.passwordField);
+    }
+
+    private void initListeners() {
+        this.adminUsernameTextField.getDocument().addDocumentListener(generateAdminUsernameListener());
+    }
+
+    private DocumentListener generateAdminUsernameListener() {
+        return new TextDocumentListenerAdapter() {
+            @Override
+            public void onDocumentChanged() {
+                if (!adminUsernameTextField.isValueInitialized()) {
+                    adminUsernameTextField.setValueInitialized(true);
+                }
+            }
+        };
+    }
+
+    @Override
+    public void setVisible(boolean visible) {
+        super.setVisible(visible);
+        rootPanel.setVisible(visible);
+    }
+
+    @Override
+    public SqlServerConfig getData() {
+        config.setServerName(serverNameTextField.getText());
+        config.setAdminUsername(adminUsernameTextField.getText());
+        config.setPassword(passwordField.getPassword());
+        config.setConfirmPassword(confirmPasswordField.getPassword());
+        return config;
+    }
+
+    @Override
+    public void setData(SqlServerConfig data) {
+        if (StringUtils.isNotBlank(config.getServerName())) {
+            serverNameTextField.setText(config.getServerName());
+        }
+        if (StringUtils.isNotBlank(config.getAdminUsername())) {
+            adminUsernameTextField.setText(config.getAdminUsername());
+        }
+        if (config.getPassword() != null) {
+            passwordField.setText(String.valueOf(config.getPassword()));
+        }
+        if (config.getConfirmPassword() != null) {
+            confirmPasswordField.setText(String.valueOf(config.getConfirmPassword()));
+        }
+    }
+
+    @Override
+    public List<AzureFormInput<?>> getInputs() {
+        final AzureFormInput<?>[] inputs = {
+            serverNameTextField,
+            adminUsernameTextField,
+            passwordFieldInput,
```

Comment:
in-place code

        return Arrays.asList(  serverNameTextField,
            adminUsernameTextField,...);

Reply:
updated.